fdbserver/ptxn/MessageSerializer.h

```diff
@@ -209,6 +209,9 @@ class TLogSubsequencedMessageSerializer {
 // Each serializer servers a single storage team ID. The serializer will only seriaize one version
 // (SubsequencedMessageSerializer supports serializing multiple versions).
 // This is useful for CommitProxies to serialize one commit into multiple storage team IDs.
+// TODO: introduce TLogGroupID into serializer for a two level serialization:
+// 		tloggroup1 : version, [team 1 header], team 1 mutations, [team 2 header], team 2 mutations
+//      tloggroup2 : version, [team 1 header], team 1 mutations, [team 3 header], team 3 mutations
```

Comment:
I think another choice is when we send the serialized data, we do not send a single StringRef, but multiple StringRefs, each StringRef represents the data for a given StorageTeamID. The issue is that, if we want to merge all serialized data into a single one, we need to do a bunch of memory copies since we do not know the length of each section of mutations. On the other hand, the TLog will have to deserialize the data in order to distribute them to different Storage Servers. Changing the TLogCommitRequest to have

```
Arena arena;
Map<std::pair<StorageTeamID, StringRef>> data;
```
(here arena might need to be replaced by an array of arenas, or depend on multiple arenas, need to figure out)
or so could simplify the issue.

Reply:
I think `Map<std::pair<StorageTeamID, StringRef>> data;` makes sense since we don't really need a big serialized string per tlog group for demo. Will do that in another PR.